azurerm/internal/services/keyvault/key_vault_key_encrypt_resource.go

```diff
@@ -0,0 +1,94 @@
+package keyvault
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/2016-10-01/keyvault"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmKeyVaultKeyEncrypt() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKeyVaultKeyEncryptCreate,
+		Read:   schema.Noop,
+		Delete: schema.Noop,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			return fmt.Errorf("resource `azurerm_key_vault_key_encrypt` does not support `import` operation")
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			// avoid test fail, add timeout config for read and delete
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key_vault_key_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateKeyVaultChildId,
+			},
+
+			"plaintext": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"algorithm": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(keyvault.RSA15),
+					string(keyvault.RSAOAEP),
+					string(keyvault.RSAOAEP256),
+				}, false),
+			},
+
+			"cipher_text": {
```

Comment:
should this also then be
```suggestion
			"encrypted_data_in_base64": {
```

Reply:
the result format is `base64url`, which is slightly different from `base64`. please refer to RFC: https://tools.ietf.org/html/rfc4648#section-5. I have noted it in the doc. Should I still rename this property?

Reply:
yes lets rename it to match what is expected so people don't have to look at docs and can tell from the property name. as for url in there is this different then any other base64 property for azure?

Reply:
lets go with `encrypted_data_in_base64` and then add a validation function to ensure its the right base64 format